lib/runAll.js

```diff
@@ -74,26 +76,15 @@ const runAll = async (
 
   const ctx = getInitialState({ quiet })
 
-  const { gitDir, gitConfigDir } = await resolveGitRepo(cwd)
-  if (!gitDir) {
-    if (!quiet) ctx.output.push(NOT_GIT_REPO)
-    ctx.errors.add(GitRepoError)
-    throw createError(ctx)
-  }
+  // here could be some factory, by default returning git workflow.
+  // repo checks and paths (gitDir, gitConfigDir) belongs to vcsWorkflow entity.
+  const vcsWorkflow = await makeVcsWorkflow({ allowEmpty, matchedFileChunks }, symbols);
```

Comment:
I think we could rename Workflow to Adapter, at least I like the sound of it. Let's make them classes so we can use the `new` keyword here:

```js
const workflow = new Adapter({ symbols })
```

Should we pass a methods instead of plain symbols? What do you use?

Reply:
> Sould we pass a collection of necessary methods instead of plain symbols?

You mean makeSomeStageError? As I understand it, characters are only needed for common mistakes.

Anyway, the methods will be more clear